spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -119,6 +157,62 @@ public Subscriber subscribe(String subscription, MessageReceiver messageHandler)
 		return subscriber;
 	}
 
+	/**
+	 * Pulls messages synchronously, on demand, using the pull request in argument.
+	 *
+	 * <p>
+	 * This method acknowledges all received messages.
+	 * @param pullRequest pull request containing the subscription name
+	 * @return the list of {@link PubsubMessage} containing the headers and payload
+	 */
+	private List<PubsubMessage> pull(PullRequest pullRequest) {
+		Assert.notNull(pullRequest, "The pull request cannot be null.");
+		Assert.notNull(subscriberFactory.getSubscriptionAdminSettings(),
+				"The SubscriptionAdminSettings cannot be null.");
+
+		try {
+			try (SubscriberStub subscriber = GrpcSubscriberStub.create(
```

Comment:
I wonder if this can go to the `SubscriberFactory` as well.

Reply:
Yeah, I guess.